### PR TITLE
Deprecate tophat kernel. Warn about flux not being conserved with some kernels

### PR DESCRIPTION
**Pull request by mcara**
This PR deprecates the "tophat" kernel which never worked correctly and adds warnings about some kernels ("gaussian" and "lanczos*") not conserving flux in the presence of distortions (non-linear terms) in input image WCS.

### CODE REVIEW

**Comment by mcara** [comment]
Regression tests running here: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1208/

File: src/cdrizzleapi.c [hunk 1]
```diff
@@ -182,6 +183,22 @@ tdriz(PyObject *obj UNUSED_PARAM, PyObject *args, PyObject *keywords)
...
+            "Kernel '%s' has been deprecated and it will be removed in a future release.",
+            kernel_str) < 1) {
+          strcpy(warn_msg, "Selected kernel has been deprecated and it will be removed in a future release.");
+      }
+      PyErr_WarnEx(PyExc_DeprecationWarning, warn_msg, 1);
+  } else if (kernel == kernel_gaussian || kernel == kernel_lanczos2 || kernel == kernel_lanczos3) {
+      if (sprintf(warn_msg,
+            "Kernel '%s' is not a flux-conserving kernel.",
+            kernel_str) < 1) {
+          strcpy(warn_msg, "Selected kernel '%s' is not a flux-conserving kernel.");
```
Comment (nden) [inline src/cdrizzleapi.c:197]: I haven't written C in a long time but don't you need to pass a value for the `%s` formatting.

Comment (mcara) [reply src/cdrizzleapi.c:197]: of corse. another 🤦‍♂️

**Comment by mcara** [comment]
There are 2 regtest failures that do not appear to be related to this PR.

**Review by nden — APPROVED** [review]
I assume there will be a similar PR in jwst. Perhaps completely removing the option to use these kernels?

**Comment by perrygreenfield** [comment]
While the code doesn't mention the issue of distortions regarding the validity of flux conservation, is it really true that it requires distortion to fail flux conservation.

**Pull request merged**